nav/dataset/dataset.py

```
@@ -61,8 +101,8 @@ def yield_filenames_from_arguments(self,
         ...
 
     @abstractmethod
-    def yield_filenames_index(self,
-                              **kwargs: Any) -> Iterator[tuple[Path, Path]]:
+    def yield_image_files_index(self,
+                                **kwargs: Any) -> Iterator[ImageFiles]:
```

Ruff (0.14.0)

Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)